app/code/Magento/CatalogGraphQl/Model/Resolver/Products/DataProvider/Customer/ContextProcessor.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogGraphQl\Model\Resolver\Products\DataProvider\Customer;
+
+use Magento\Customer\Api\Data\GroupInterface;
+use Magento\CustomerGraphQl\Model\Customer\GetCustomer;
+use Magento\GraphQl\Model\Query\ContextFactoryInterface;
+use Magento\Customer\Model\Session;
+
+/**
+ * Processing applied to the collection after load
+ */
+class ContextProcessor
+{
+    /**
+     * @var ContextFactoryInterface
+     */
+    private $contextFactory;
+
+    /**
+     * @var GetCustomer
+     */
+    private $getCustomer;
+
+    /**
+     * @var Session
+     */
+    private $customerSession;
+
+    /**
+     * ContextProcessor constructor.
+     * @param ContextFactoryInterface $contextFactory
+     * @param GetCustomer $getCustomer
+     * @param Session $customerSession\
+     */
+    public function __construct(
+        ContextFactoryInterface $contextFactory,
+        GetCustomer $getCustomer,
+        Session $customerSession
+    ) {
+
+        $this->contextFactory = $contextFactory;
+        $this->getCustomer = $getCustomer;
+        $this->customerSession = $customerSession;
+    }
+
+    public function process()
```

Comment:
Can you check \Magento\CustomerGraphQl\Model\Context\AddUserInfoToContext, I think it is already doing something similar, but it uses request headers instead of session. we shouldn't use session directly in GraphQl. Maybe you can just add groupId as part of the context that gets passed to resolvers.

Reply:
@pmarjan Please resolve this issue when you have a chance.